### PR TITLE
Improve performance of @notifications endpoint.

### PR DESCRIPTION
**Pull request by njohner**
We now avoid getting all the users notifications from the OGDS DB, instead only getting the batch we actually want.
This should dramatically increase the performance of the `@notifications` endpoint for users with a large number of notifications (there is a user in SG with 150'000 notifications). Note that slow requests on the OGDS will block all other users from all tenants, so this is a critical fix which we will backport to our LTS and to the SG release.

For a user with 70'000 notifications, performance improvement was from 2s to 300ms.

For [CA-5939]

## Checklist
- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

[CA-5939]: https://4teamwork.atlassian.net/browse/CA-5939?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: changes/CA-5939.other [hunk 1]
```diff
@@ -0,0 +1 @@
+Improve performance of @notifications endpoint. [njohner]
```
Comment (elioschmutz) [inline changes/CA-5939.other:1]: also add it in the API-changelog

**Review by elioschmutz — APPROVED** [review]
🌟

File: changes/CA-5939.other (hunk 1 above)
Comment (njohner) [reply changes/CA-5939.other:1]: You think so? It's not really an API change IMO, so I'd rather not add that in the API-changelog.

**Comment by njohner** [comment]
@elioschmutz can you have a look at the second commit?

**Review by elioschmutz — APPROVED** [review]

**Pull request merged**